### PR TITLE
feat: Add support for data-driven InfluxQL tests

### PR DESCRIPTION
**Pull request by stuartcarnie**
Helps #6112

Teach the data-driven query testing framework how to distinguish queries of different languages, and is capable of running SQL and InfluxQL tests.

We opted to use the file extension as the discriminator for the language, where `.sql` will interpret the test file content as DataFusion SQL queries and `.influxql` as InfluxQL queries. If the need arises, it should be relatively easy to adopt additional dialects or languages in the future. 

### Summary of changes

To accomplish this, the naming of the output and expected files now retain the language extension, meaning:

1. output files to the `out` directory are now `testfile.ext.out`
2. expected files are now `testfile.ext.expected`

This change allows us to avoid conflicts, if we have two tests with the same base file name and differing extensions. For example, say we have `basic.sql` and `basic.influxql`, the output files will be named accordingly:

```
basic.influxql
basic.influxql.out
basic.influxql.expected
basic.sql
basic.sql.out
basic.sql.expected
```

The existing test output files have been updated to reflect these changes.

#### Edit-Test cycle improvement

Commit 65429983d41fe6e0d22a4618be8570bd9b73e303 adds a check to the build.rs to avoid rebuilding the test binary if nothing has changed, including the `GIT_HASH`

### CODE REVIEW

File: influxdb_iox/tests/query_tests2/cases.rs [hunk 1]
```diff
@@ -300,3 +300,19 @@ async fn different_tag_sets() {
...
+
+mod influxql {
+    use super::*;
+
+    #[tokio::test]
+    async fn issue_6112() {
+        test_helpers::maybe_start_logging();
+
+        TestCase {
+            input: "cases/in/issue_6112.influxql",
```
Comment (alamb) [inline influxdb_iox/tests/query_tests2/cases.rs:312]: 🥳

File: test_helpers_end_to_end/src/steps.rs [hunk 2]
```diff
@@ -445,6 +458,26 @@ where
                     assert_batches_sorted_eq!(expected, &batches);
                     info!("====Done running");
                 }
+                Step::InfluxQLQueryAndCompare {
```
Comment (alamb) [inline test_helpers_end_to_end/src/steps.rs:461]: Maybe as a follow on PR we can rename `Step::QueryAndCompare` to `Step::SQLQueryAndCompare` for consistency

**Review by alamb — APPROVED** [review]
I went through this code carefully and I think it is great. Thank you @stuartcarnie . FYI @carols10cents 

> Commit https://github.com/influxdata/influxdb_iox/commit/65429983d41fe6e0d22a4618be8570bd9b73e303 adds a check to the build.rs to avoid rebuilding the test binary if nothing has changed, including the GIT_HASH

🙏  thank you -- that has been bothering me

As I expect a non trivial amount of churn in the next few days in the end to end tests, I am going to merge this PR in to minimize accumulating conflicts

File: test_helpers_end_to_end/src/steps.rs (hunk 2 above)
Comment (stuartcarnie) [reply test_helpers_end_to_end/src/steps.rs:461]: Agreed 💯

**Comment by stuartcarnie** [comment]
Thanks for the review and merge, @alamb

**Pull request merged**